generic3g/specs/FrequencyAspect.F90

```diff
@@ -0,0 +1,140 @@
+#include "MAPL_Generic.h"
+module mapl3g_FrequencyAspect
+   use mapl3g_StateItemAspect
+   use mapl3g_AccumulatorActionInterface
+   use esmf
+   implicit none
+   private
+
+   public :: FrequencyAspect
+
+   type, extends(StateItemAspect) :: FrequencyAspect
+      type(ESMF_TimeInterval) :: dt
+      character(len=:), allocatable :: accumulation_type
+   contains
+      procedure :: matches
+      procedure :: supports_conversion_general
+      procedure :: supports_conversion_specific
+      procedure :: make_action
+   end type FrequencyAspect
+
+   interface FrequencyAspect
+      module procedure :: construct_frequency_aspect
+   end interface FrequencyAspect
+
+   interface operator(.divides.)
+      module procedure :: divides
+   end interface operator(.divides.)
+
+contains
+
+   function construct_frequency_aspect(dt, accumulation_type) result(aspect)
+      type(FrequencyAspect) :: aspect
+      type(ESMF_TimeInterval), optional, intent(in) :: dt
+      character(len=*), optional, intent(in) :: accumulation_type
+
+      associate(adt => aspect%dt, atype => aspect%accumulation_type)
+         atype = NO_ACCUMULATION
+         if(present(accumulation_type)) atype = accumulation_type
+         call ESMF_TimeIntervalSet(adt, ns=0)
+         if(.not. present(dt)) return
+         if(is_zero(dt)) return
+         adt = dt
+      end associate
+      
+   end function construct_frequency_aspect
+
+   logical function matches(this, aspect) result(does_match)
+      import :: StateItemAspect
+      class(FrequencyAspect), intent(in) :: this
+      class(StateItemAspect), intent(in) :: aspect
+
+      does_match = .FALSE.
+      if(is_zero(this%dt)) return
+      select type(StateItemAspect)
+      class is (FrequencyAspect)
+         if(is_zero(aspect%dt)) return
+         does_match = this%dt == aspect%dt
+      class(default)
+         does_match = .FALSE.
```

Comment:
Default clause not needed here due to the initial assignment above.
```suggestion
```

Reply:
done